### PR TITLE
Added test for gmm_train

### PR DESCRIPTION
**Pull request by gauravcr7rm**
This is the test for gmm_train() . I am facing problems for creating test for using kmeans in the same.

### CODE REVIEW

File: src/mlpack/tests/gmm_train.cpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
+#include<string>
```
Comment (zoq) [inline src/mlpack/tests/gmm_train.cpp]: Can you take a look at the existing tests and adapt the style. https://github.com/mlpack/mlpack/wiki/DesignGuidelines should be helpful as well.

File: src/mlpack/tests/gmm_train.cpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  	SetInputParam("input", std::move(inputData));
+  	SetInputParam("refined_start", true);
+  	SetInputParam("clusters", c);
+  	SetInputParam("percentage", std::move(P));     // Invalid
+
+  	Log::Fatal.ignoreInput = true;
+  	BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  	Log::Fatal.ignoreInput = false;
+}
+
```
Comment (zoq) [inline src/mlpack/tests/gmm_train.cpp]: Picky style comment, can you remove the extra lines here.

File: src/mlpack/tests/gmm_train.cpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  	Log::Fatal.ignoreInput = false;
+
+}
+ 
+ /**
+ * Checking that percentage is between 0 and 1 when --refined_start is specified
+*/
+BOOST_AUTO_TEST_CASE(RefinedStartPercentageTest)
+{
+  	int g = 3;
```
Comment (zoq) [inline src/mlpack/tests/gmm_train.cpp]: Tabs should be two spaces wide; see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#tabbing for more information.

File: src/mlpack/tests/gmm_train.cpp (hunk 1 above)
Comment (gauravcr7rm) [reply src/mlpack/tests/gmm_train.cpp]: Ok i'll rectify the code by following the design guideline.

**Comment by gauravcr7rm** [comment]
@zoq please tell me if any other changes are required.

**Comment by gauravcr7rm** [comment]
@rcurtin i have made all the changes and have included more checks for better coverage.Please tell me if anything more can be done in here.

**Comment by rcurtin** [comment]
Can you rebase this PR against the current master so we can see if it builds?  Thanks!
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request may be helpful if you haven't done this before.

**Comment by gauravcr7rm** [comment]
sorry for the delay,my exams were going on.
I've rebased the pr as instructed by you and also have the made the changes in probabilty test and generate test also.
Please have look @rcurtin

**Comment by rcurtin** [comment]
There is still a merge conflict; can you please resolve that so that the CI tests run?  Thanks.

**Comment by gauravcr7rm** [comment]
@rcurtin ci test heve passed and the conflict is resolved..please have a look

**Comment by rcurtin** [comment]
This is also solved by #1899.

**Pull request closed without merging**